provides.py

```diff
@@ -37,12 +37,12 @@ def joined_or_changed(self):
 
     @hook('{provides:kube-control}-relation-{broken,departed}')
     def departed(self):
-        """Remove all states.
+        """Remove all states when the last remote unit leaves the relation.
 
         """
-        conv = self.conversation()
-        conv.remove_state('{relation_name}.connected')
-        conv.remove_state('{relation_name}.gpu.available')
+        if not self.conversations():
```

Comment:
In an `@hook` context, there will always be exactly one conversation, for the remote unit of the hook.  I think that this interface layer is actually correct (save for the `-broken` hook issue mentioned in the other comment) and the issue may be in how it's being used in the charm layer.